packages/transactions/src/instructions.ts

```diff
@@ -39,8 +18,10 @@ export function prependTransactionInstruction<TTransaction extends BaseTransacti
     instruction: TTransaction['instructions'][number],
     transaction: TTransaction | (TTransaction & ITransactionWithSignatures)
 ): TTransaction | Omit<TTransaction, keyof ITransactionWithSignatures> {
-    const nextInstructions = [instruction, ...transaction.instructions] as const;
-    const out = replaceInstructions(transaction, nextInstructions);
+    const out = {
+        ...getUnsignedTransaction(transaction),
+        instructions: [instruction, ...transaction.instructions],
+    };
     Object.freeze(out);
     return out;
 }
```

Comment:
Assuming you offer `prepend` and `append` to add the nonce ixs before or after a set of arbitrary instructions?

Reply:
In order for a transaction to make use of a nonce:

1. The nonce has to be set in place of the ‘recent blockhash’
2. The first instruction in the transaction _has_ to be the `AdvanceNonceAccount` instruction

So prepend only. No appending allowed.

https://docs.solana.com/implemented-proposals/durable-tx-nonces#runtime-support

Reply:
In any case, these lines weren't part of this PR.